### PR TITLE
Fixes mmis not working properly with mechas

### LINKED ISSUE #23718: MMIs in mechs can't control the mech
They're not able to move around and don't get the mech UI like they're supposed to. They are basically stuck and can't do anything but talk for the rest of the round.

### PR DESCRIPTION
**Pull request by lzimann**
:cl: Lzimann
fix: MMIs/posibrains works with mechas again
/:cl:

Fixes #23718

This part of the code mentally exhausted me.

### CODE REVIEW

File: code/game/mecha/mecha.dm [hunk 1]
```diff
@@ -25,7 +25,7 @@
 	force = 5
 	flags = HEAR
 	var/can_move = 1
-	var/mob/living/carbon/occupant = null
+	var/mob/living/carbon/human/occupant = null
```
Comment (lzimann) [inline code/game/mecha/mecha.dm]: This was accidental(althought the type here really doesn't matter except when trying to grab the DNA).

File: code/game/mecha/mecha.dm [hunk 2]
```diff
@@ -358,8 +358,12 @@
...
 			else
 				occupant.clear_alert("mech damage")
-
-		if(occupant.loc != src) //something went wrong
+		var/actual_loc = occupant.loc
+		if(istype(actual_loc, /obj/item/device/mmi))
+			var/obj/item/device/mmi/M = actual_loc
+			actual_loc = M.mecha
+		if(actual_loc != src) //something went wrong
+			world << "occupant.loc"
```
Comment (Cyberboss) [inline code/game/mecha/mecha.dm]: .

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by Cyberboss** [comment]
Maybe I didn't cause this bug... This is way more complicated than what I touched

**Comment by PKPenguin321** [comment]
Fixes https://github.com/tgstation/tgstation/issues/23718

File: code/game/mecha/mecha.dm (hunk 2 above)
Comment (lzimann) [inline code/game/mecha/mecha.dm]: i just added to see if it was this check that was triggering

**Pull request merged**